state/syncer/userAccountsSyncer.go

```diff
@@ -325,6 +333,29 @@ func (u *userAccountsSyncer) resetTimeoutHandlerWatchdog() {
 	u.timeoutHandler.ResetWatchdog()
 }
 
+// MissingDataTrieNodeFound is called whenever a missing data trie node is found.
+// This will trigger the sync process for the whole sub trie, starting from the given hash.
+func (u *userAccountsSyncer) MissingDataTrieNodeFound(hash []byte) {
+	u.mutex.Lock()
+	defer u.mutex.Unlock()
+
+	u.timeoutHandler.ResetWatchdog()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer func() {
+		u.cacher.Clear()
+		cancel()
+	}()
+
+	_, err := u.createAndStartSyncer(ctx, hash)
+	if err != nil {
+		log.Error("cannot sync trie", "err", err, "hash", hash)
+		return
+	}
+
```

Comment:
should we udpate data trie statistics here also?

Reply:
Yes, added.